sys/mips/timer.c

```diff
@@ -85,10 +100,13 @@ static int mips_timer_stop(timer_t *tm) {
 static bintime_t mips_timer_gettime(timer_t *tm) {
   device_t *dev = tm->tm_priv;
   mips_timer_state_t *state = dev->state;
-  uint64_t count = read_count(state);
-  uint32_t sec = count / tm->tm_frequency;
-  uint32_t frac = count % tm->tm_frequency;
-  bintime_t bt = bintime_mul(HZ2BT(tm->tm_frequency), frac);
+  uint32_t sec, ticks;
+  WITH_INTR_DISABLED {
```

Comment:
the interrupts are disabled to avoid change of time?

Reply:
We want the time read to be atomic.